alvr/server/cpp/platform/win32/VideoEncoderNVENC.cpp

```diff
@@ -93,6 +97,31 @@ void VideoEncoderNVENC::Shutdown()
 
 void VideoEncoderNVENC::Transmit(ID3D11Texture2D *pTexture, uint64_t presentationTime, uint64_t frameIndex, uint64_t frameIndex2, uint64_t clientTime, bool insertIDR)
 {
+	if (m_enableAdaptiveBitrate && m_Listener) {
+		uint64_t latencyUs = m_Listener->GetStatistics()->GetSendLatencyAverage();
+		if (latencyUs != 0) {
+			if (latencyUs > m_adaptiveBitrateTarget + m_adaptiveBitrateThreshold) {
+				m_bitrateInMBits -= 3;
+			} else if (latencyUs < m_adaptiveBitrateTarget - m_adaptiveBitrateThreshold) {
+				m_bitrateInMBits += 1;
+			}
+			if (m_bitrateInMBits > m_adaptiveBitrateMaximum) {
+				m_bitrateInMBits = m_adaptiveBitrateMaximum;
+			} else if (m_bitrateInMBits < 5) {
+				m_bitrateInMBits = 5;
```

Comment:
Adaptative bitrate tuning should be in its own class, preferably outside of platform-specific code.
It could be a member of Statistics class, or a member of the encoder, and that would help making the API clear: what it takes as input to make decisions, and what is the output.